src/XmlSchemaValidator.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\XmlValidator;
+
+use DOMDocument;
+
+final class XmlSchemaValidator implements XmlValidator
+{
+    use LibXmlValidator;
+
+    private $schema;
+
+    public function __construct(string $schema)
```

Comment:
Not sure what the underlying extension accepts, but this could easily be be interpreted as a string containing the whole XML schema document; `$schema_filename`, `$schema_path`?